Sources/HAP/Base/Accessory.swift

```diff
@@ -30,6 +35,29 @@ open class Accessory {
     public let type: AccessoryType
     public let info: Service.Info
     internal let services: [Service]
+    
+    // An accessory implementation can set this flag to false if a device becomes unreachable for a period
+    // If the accessory has a BridgingState Service, then the reachable property of that Service is set
+    //
+    open var reachable = true {
+        didSet {
+            if self.reachable != oldValue,
+               let bridgingStateService = services.first(where: { $0.type == .bridgingState }) as! Service.BridgingState? {
+                bridgingStateService.reachable.value = self.reachable
+            }
+        }
+    }
+    
+    // An accessory must provide a text identifier, which is guranteed to be unique.
```

Comment:
We can rely on the serial number, and thus require the serial number to be unique (as it should be).

Reply:
Can do, but in that case we should change the init for the Service.Info class to remove the default parameter for serial number 

    serialNumber: String = "undefined"

and perhaps add a check in Device to ensure there are no duplicate serial numbers.

Reply:
Yeah we would need the user to explicitly specify a serial number.

Reply:
I have renamed the property in Accessory to 'uniqueSerialNumber', and by default look for it from the info record. An implementation can either set the info record or override the property. 

In Device.addAccessories, each Accessory is checked against all Accessories previously added to ensure the new device has a unique serial number. If not, it is not added and a message sent to logger. 

I prefer not to use a precondition, fatal error or exception because implementations may be adding devices by network discovery, and for whatever reason may end up with a two devices with the same serial number, so in such a case I would not want the app to crash.

Reply:
We should use a trap on invalid API usage. Logging an error is not visible enough to the consumer. Also printing a log message does not trigger some programming logic on the consumer which can be used to resolve this error.

I understand it could possible that duplicate serial numbers might occur. However the consumer should ensure that they are unique, thus could query the `Device` to see if there's already an accessory with that serial number. If there's already an accessory with that serial number, the consumer should decide what to do; either ensure the serial number is unique, or not add the accessory at all.

Reply:
OK, I have revised Device.addAccessories to return an array of Accessories that were successfully added, and also provided a function canAddAccessory() which can be used to check beforehand. I've also added a couple of preconditions to addAccessory, to check the device is a bridge and that the upper limit of 99 accessories (100 including the bridge) will not be exceeded.